PWGUD/Core/UPCJpsiCentralBarrelCorrHelper.h

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief
+/// \author Sara Haidlova, sara.haidlova@cern.ch
+/// \since March 2024
+
+#ifndef PWGUD_CORE_UPCJPSICENTRALBARRELCORRHELPER_H_
+#define PWGUD_CORE_UPCJPSICENTRALBARRELCORRHELPER_H_
+
+#include <algorithm>
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace std;
+
+enum ParticleType {
+  P_ELECTRON = 0,
+  P_MUON = 1,
+  P_PROTON = 2
+};
+
+template <typename T>
+int testPIDhypo(T trackPID)
+{
+  float nSigmaTPC[3];
+  nSigmaTPC[P_ELECTRON] = std::abs(trackPID.tpcNSigmaEl());
+  nSigmaTPC[P_MUON] = std::abs(trackPID.tpcNSigmaMu());
+  nSigmaTPC[P_PROTON] = std::abs(trackPID.tpcNSigmaPr());
+  int enumChoiceTPC = std::distance(std::begin(nSigmaTPC),
+                                    std::min_element(std::begin(nSigmaTPC), std::end(nSigmaTPC)));
+  if (trackPID.hasTPC()) {
+    return enumChoiceTPC;
+  } else {
+    return -1;
+  }
+}
+
+float mom(float px, float py, float pz)
+{
+  return std::sqrt(px * px + py * py + pz * pz);
+}
+
+float phi(float px, float py)
+{
+  if (px != 0) {
+    return std::atan2(py, px);
+  } else {
+    return -999;
+  }
+}
+
+float eta(float px, float py, float pz)
+{
+  float mom = std::sqrt(px * px + py * py + pz * pz);
+  if (mom != 0) {
+    return std::atanh(pz / mom);
+  } else {
+    return -999;
+  }
+}
```

Comment:
All these and many other calculation in your code are already implemented in the `RecoDecay` class and can be performed with `std::array` objects which are much lighter than `TLorentzVector`.